paparazzi/src/main/java/app/cash/paparazzi/Environment.kt

```diff
@@ -110,3 +111,33 @@ private fun checkInstalledJvm() {
     )
   }
 }
+
+internal val OsLabel: String by lazy {
+  val osName = System.getProperty("os.name").lowercase(Locale.US)
+  when {
+    osName.startsWith("windows") -> "win"
+    osName.startsWith("mac") -> {
+      // System.getProperty("os.arch") returns the os of the jre, not necessarily of the platform we are running on,
+      // try /usr/bin/arch to get the actual architecture
+      val osArch = ProcessBuilder("/usr/bin/arch")
+        .start()
+        .inputStream
+        .bufferedReader()
+        .lines()
+        .findFirst()
+        .getOrNull()
+        ?.lowercase(Locale.US)
+        ?.let {
+          if (it.startsWith("i386")) "mac" else "mac-arm"
+        }
+      val jreArch = System.getProperty("os.arch").lowercase(Locale.US).let {
+        if (it.startsWith("x86")) "mac" else "mac-arm"
+      }
+      if (osArch != null && osArch != jreArch) {
+        System.err.println("Mismatched architectures, os is:$osArch but the JRE is running with:$jreArch. Installing the correct JDK for your architecture will improve performance.")
+      }
+      osArch ?: jreArch
```

Comment:
```suggestion
            when (osArch) {
                null -> jreArch
                jreArch -> osArch
                else -> throw RuntimeException("Mismatched architectures, os is:$osArch but the JRE is running with:$jreArch. Please installing the correct JDK for your architecture.")
            } 
```

I propose making this fail fast/crash, given the error in #1031 seemed to make this a blocker for others who accidentally installed the wrong JDK.